### ISSUE TITLE
curly: multi-or-nest option reports an error for a single-statment block with a semicolon on the next line

### ISSUE THREAD

**Issue by cherryblossom000** [eslint/eslint#12370]
**Tell us about your environment**
* **ESLint Version:** 6.5.1
* **Node Version:** 12.11.0
* **npm Version:** 6.11.3

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```json
{
  "rules": {
    "curly": [2, "multi-or-nest"]
  }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

I used the following code in `file.js`:
```js
if (true) console.log('some statement')
;(function () {})()
```

The issue occurs whenever the statement in the if block (or any block) has the semicolon on the following line.

Then I linted it:
```bash
eslint file.js
```

**What did you expect to happen?**
No error would occur as there is only 1 statement in the if block.

**What actually happened? Please include the actual, raw output from ESLint.**
I got the following error:
```
file.js
  1:1  error  Expected { after 'if' condition  curly

✖ 1 problem (1 error, 0 warnings)
  1 error and 0 warnings potentially fixable with the `--fix` option.
```

When fixing it with `eslint --fix file.js`, it gets changed to
```js
if (true) {console.log('some statement')
;}(function () {})()
```
which conflicts with the `"semi": [2, "never"]` rule as that would change it to:
```js
if (true) {console.log('some statement')
}(function () {})()
```
which would cause `curly` to report `Unnecessary { after 'if' condition`.

**Are you willing to submit a pull request to fix this bug?**
Yes.

**Comment by cherryblossom000** [comment]
I think the issue is here:

https://github.com/eslint/eslint/blob/c47fa0dfc76211b3b0e5649c63acdd9606ce0eca/lib/rules/curly.js#L99-L104

`last` would be the semicolon, which is on a different line to the first token in
```js
console.log('some statement')
;
```

**Comment by kaicataldo** [comment]
Verified using our [demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IFwiY3VybHlcIjogWzIsIFwibXVsdGktb3ItbmVzdFwiXSAqL1xuXG5pZiAodHJ1ZSkgY29uc29sZS5sb2coJ3NvbWUgc3RhdGVtZW50JylcbjsoZnVuY3Rpb24gKCkge30pKCkiLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjYsInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnt9LCJlbnYiOnt9fX0=). Nice find! Definitely seems like we should use the previous token for this check in this case.

**Comment by cherryblossom000** [comment]
I fixed it in #12378.
